src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -585,17 +629,116 @@ def set_parameter_of_bus(  # pylint: disable=too-many-locals, too-many-statement
+                )
+            else:
+                settings_fem = next(fem for fem in settings_controllers["fems"] if fem["fem"] == con_fem)
+                settings_offset = next(
+                    analog_output
+                    for analog_output in settings_fem["analog_outputs"]
+                    if analog_output["port"] == con_port
+                )
+            settings_offset["offset"] = dc_offset
+            if self._config_created:
+                if con_fem is None:
+                    self._config["controllers"][con_name]["analog_outputs"][con_port]["offset"] = dc_offset  # type: ignore[typeddict-item]
+                else:
+                    self._config["controllers"][con_name]["fems"][con_fem]["analog_outputs"][con_port][  # type: ignore[typeddict-item]
+                        "offset"  # type: ignore[typeddict-unknown-key]
+                    ] = dc_offset
+            if self._is_connected_to_qm:
+                self._qm.set_output_dc_offset_by_element(element=bus, input="single", offset=dc_offset)
+            return
+
+        if parameter in [Parameter.OFFSET_I, Parameter.OFFSET_Q]:
+            key = "I" if parameter in Parameter.OFFSET_I else "Q"
+            con_name, con_port, con_fem = self.get_controller_from_element(element=element, key=key)
+            input_offset = float(value)
+            settings_controllers = next(
+                controller for controller in self.settings.controllers if controller["name"] == con_name
+            )
+            if con_fem is None:
+                settings_offset = next(
+                    analog_output
+                    for analog_output in settings_controllers["analog_outputs"]
+                    if analog_output["port"] == con_port
+                )
+            else:
+                settings_fem = next(fem for fem in settings_controllers["fems"] if fem["fem"] == con_fem)
+                settings_offset = next(
+                    analog_output
+                    for analog_output in settings_fem["analog_outputs"]
+                    if analog_output["port"] == con_port
+                )
+            settings_offset["offset"] = input_offset
+            if self._config_created:
+                if con_fem is None:
+                    self._config["controllers"][con_name]["analog_outputs"][con_port]["offset"] = input_offset  # type: ignore[typeddict-item]
+                else:
+                    self._config["controllers"][con_name]["fems"][con_fem]["analog_outputs"][con_port][  # type: ignore[typeddict-item]
+                        "offset"  # type: ignore[typeddict-unknown-key]
+                    ] = input_offset
+            if self._is_connected_to_qm:
+                self._qm.set_output_dc_offset_by_element(element=bus, input=key, offset=input_offset)
+            return
+
+        if parameter in [Parameter.OFFSET_OUT1, Parameter.OFFSET_OUT2]:
+            output = "out1" if parameter in Parameter.OFFSET_OUT1 else "out2"
```

Comment:
This should be equals (==) right?
```suggestion
            output = "out1" if parameter == Parameter.OFFSET_OUT1 else "out2"
```